feat(oxfmt): use `prettier` directly and bundle `prettier`

Dunqing opened the pull request:
`@prettier/sync` is the synchronous version of `prettier`, which uses the `Worker` to hack it and make it synchronous. 

Everything worked fine before we wanted to bundle them together. However, it hangs when they are bundled. I haven't found the reason why it hangs. Anyway, I don't think `@prettier/sync` is a good choice because it is not `real` synchronous, which sacrifices some performance to make it available. 

Note: I am not familiar with NAPI, so I'm not sure whether the API is used correctly.

Dunqing commented:
* **#15544** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15544?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15544?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15544).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

leaysgur commented on apps/oxfmt/src-js/embedded.ts:
```diff
@@ -25,27 +25,21 @@ const TAG_TO_PARSER: Record<string, string> = {
-    return code;
-  }
+    })
+    .then((formatted) => formatted.trimEnd());
```
I think this is valid. We should `catch` here.

leaysgur commented:
I think this would be ideal if there were no drawbacks.

Let me try, run benchmark locally.

leaysgur reviewed (approved):
LGTM.

Now we can remove `dependencies` field from `npm/oxfmt/package.json`.

https://github.com/oxc-project/oxc/blob/f4b75b631c069decef11adee0435765e0811779e/npm/oxfmt/package.json#L27

Dunqing committed:
feat(oxfmt): use `prettier` directly and bundle `prettier` (#15544)

`@prettier/sync` is the synchronous version of `prettier`, which uses the `Worker` to hack it and make it synchronous.

Everything worked fine before we wanted to bundle them together. However, it hangs when they are bundled. I haven't found the reason why it hangs. Anyway, I don't think `@prettier/sync` is a good choice because it is not `real` synchronous, which sacrifices some performance to make it available.

Note: I am not familiar with NAPI, so I'm not sure whether the API is used correctly.
